src/ls/javascript/L8/1_Retry.js

```
@@ -0,0 +1,24 @@
+class MultiplicatorUnitFailure extends Error {}
+
+function primitiveMultiply(a, b) {
+  if (Math.random() < 0.2) {
+    return a * b;
+  } else {
+    throw new MultiplicatorUnitFailure("Klunk");
+  }
+}
+
+function reliableMultiply(a, b) {
+  for (;;) {
+    try {
+      return primitiveMultiply(a, b);
+    } catch (e) {
+      if (!(e instanceof MultiplicatorUnitFailure)) {
```

Refactor this:
(1) Is expected that the condition inside the if() statement gets a false value every time, so, you are not catching the errors that are an instanceof MultiplicatorUnitFailure
(2) Inside this scope you are throwing the error, that blocks the node operation.

Refactor it to log the error catch by the promise management.

---

MultiplicatorUnitFailure fails on average 80% of the time, the if statement, on line 16, is filtering those errors allowing MultiplicatorUnitFailure to be called again. It will show, or re-throw, an error only if it does not come from MultiplicatorUnitFailure. I understand that this is what is indicated in the exercise statement. "...The corresponding catch block should rethrow the exception when it is not an instance of MultiplicatorUnitFailure and ensure the call is retried when it is...."